homeassistant/helpers/config_validation.py

```diff
@@ -745,11 +739,20 @@ def deprecated(
             " please remove it from your configuration"
         )
 
+    if fail:
+        keyword_style_adapter = KeywordStyleAdapter(
+            logging.getLogger(module_name)
+        ).error
+    else:
+        keyword_style_adapter = KeywordStyleAdapter(
+            logging.getLogger(module_name)
+        ).warning
+
     def validator(config: dict) -> dict:
-        """Check if key is in config and log warning."""
+        """Check if key is in config and log warning or error."""
         if key in config:
             try:
```

Comment:
It's the `replace` which may throw, maybe we could clean this up to something like:
```py
            try:
                warning_local = warning.replace(
                    "'{key}' option",
                    f"'{key}' option near {config.__config_file__}:{config.__line__}",  # type: ignore
                )
            except AttributeError:
                warning_local = warning
            warning_local = warning_local.format(
                key=key,
                replacement_key=replacement_key,
            )

            if fail:
                raise vol.Invalid(warning_local)

           keyword_style_adapter...
```

Reply:
done